astropy/stats/spatial.py

```diff
@@ -0,0 +1,314 @@
+                dimensions of the rectangular area of study. It assumes that
+                all the points lie in a bounded rectangular region satisfying
+                x_min < x_i < x_max; y_min < y_i < y_max. A detailed
+                description of this method can be found on ref [4].
+            * 'ohser'
+                this method uses the isotropized set covariance function of
+                the window of study as a weigth to correct for
+                edge-effects. A detailed description of this method can be
+                found on ref [4].
+            * 'var-width'
+                this method considers the distance of each observed point to
+                the nearest boundary of the study window as a factor to
+                account for edge-effects. See [3] for a brief description of
+                this method.
+            * 'ripley'
+                this method is known as Ripley's edge-corrected estimator.
+                The weight for edge-correction is a function of the
+                proportions of circumferences centered at each data point
+                which crosses another data point of interest. See [3] for
+                a detailed description of this method.
+
+        Returns
+        -------
+        ripley : 1D array
+            Ripley's K function estimator evaluated at ``radii``.
+        """
+
+        data = np.asarray(data)
+
+        if not data.shape[1] == 2:
+            raise ValueError('data must be an n by 2 array, where n is the '
+                             'number of observed points.')
+
+        npts = len(data)
+        ripley = np.zeros(len(radii))
+
+        if mode == 'none':
+            distances = self._pairwise_distances(data)
+            for r in range(len(radii)):
+                ripley[r] = (distances < radii[r]).sum()
+
+            ripley = self.area * 2. * ripley / (npts * (npts - 1))
+        # eq. 15.11 Stoyan book page 283
+        elif mode == 'translation':
+            diff = self._pairwise_diffs(data)
+            distances = self._pairwise_distances(data)
+            intersec_area = (((self.x_max - self.x_min) - diff[:][:, 0]) *
+                             ((self.y_max - self.y_min) - diff[:][:, 1]))
+
+            for r in range(len(radii)):
+                dist_indicator = distances < radii[r]
+                ripley[r] = ((1 / intersec_area) * dist_indicator).sum()
+
+            ripley = (self.area**2 / (npts * (npts - 1))) * 2 * ripley
+        # Stoyan book page 123 and eq 15.13
+        elif mode == 'ohser':
+            distances = self._pairwise_distances(data)
+            a = self.area
+            b = max((self.y_max - self.y_min) / (self.x_max - self.x_min),
+                    (self.x_max - self.x_min) / (self.y_max - self.y_min))
+            x = distances / math.sqrt(a / b)
+            u = np.sqrt((x * x - 1) * (x > 1))
+            v = np.sqrt((x * x - b ** 2) * (x < math.sqrt(b**2 + 1)) * (x > b))
```

Comment:
Until now I assumed you did `x * x` because it could be more efficient than `x**2` but in this line you do both `x * x - b ** 2`, maybe stick to one style? If you really want to improve the performance here you could calculate some calculations up front:

```
x_squared = x*x
b_squared = b*b
x_greater_than_b = x > b
```

because these are reused multiple times. But it shouldn't make to much of a difference here. :)